src/app/Library/CrudPanel/Traits/SaveActions.php

```diff
@@ -123,32 +332,94 @@ public function performSaveAction($itemId = null)
                 'success'      => true,
                 'data'         => $this->entry,
                 'redirect_url' => $redirectUrl,
+                'referrer_url' => $referrer_url,
             ];
         }
 
+        if (isset($referrer_url)) {
+            session()->flash('referrer_url_override', $referrer_url);
+        }
+
         return \Redirect::to($redirectUrl);
     }
 
     /**
-     * Get the translated text for the Save button.
+     * Returns the default Backpack save actions for cruds.
      *
-     * @param string $actionValue
+     * @return array
+     */
+    public function setupBackpackDefaultSaveActions()
+    {
+        $defaultSaveActions = [
+            [
+                'name' => 'save_and_back',
+                'permissions' => function ($crud) {
+                    return $crud->hasAccess('list');
+                },
+                'redirect' => function ($crud, $request, $itemId = null) {
+                    return $request->has('http_referrer') ? $request->get('http_referrer') : $crud->route;
+                },
+                'button_text' => trans('backpack::crud.save_action_save_and_back'),
+                'order' => 2,
+            ],
+            [
+                'name' => 'save_and_edit',
+                'permissions' => function ($crud) {
+                    return $crud->hasAccess('update');
+                },
+                'redirect' => function ($crud, $request, $itemId = null) {
+                    $itemId = $itemId ?: $request->input('id');
+                    $redirectUrl = $crud->route.'/'.$itemId.'/edit';
+                    if ($request->has('locale')) {
+                        $redirectUrl .= '?locale='.$request->input('locale');
+                    }
+                    if ($request->has('current_tab')) {
+                        $redirectUrl = $redirectUrl.'#'.$request->get('current_tab');
+                    }
+
+                    return $redirectUrl;
+                },
+                'referrer_url' => function ($crud, $request, $itemId) {
```

Comment:
@pxpm ping here - you said you'd give more details please?

Reply:
This allows us to fix #2328. 

In the performSaveAction() I added: 
```
//allow the save action to define default http_referrer (url for the save_and_back button)
            if (isset($this->availableSaveActions[$saveAction]['referrer_url'])) {
                if (is_callable($this->availableSaveActions[$saveAction]['referrer_url'])) {
                    $referrer_url = $this->availableSaveActions[$saveAction]['referrer_url']($this, $request, $itemId);
                }
```
So when you click `save_and_edit` we setup http_referrer to the one we want, not the default crud referrer.

Reply:
![giphy-downsized](https://user-images.githubusercontent.com/1032474/73263251-d3c70b80-41d8-11ea-82f2-abd290464a47.gif)